error.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import "fmt"
+
+// We define a custom error type so that we can provide friendlier error messages
+type fetchError struct {
```

Comment:
I don't have a very strong preference, but I think most of the Go code I've seen so far capitalizes type names. It helps distinguish them a bit from variable names.